packages/@angular/cli/tasks/serve.ts

```diff
@@ -31,7 +31,15 @@ export default Task.extend({
     if (projectConfig.project && projectConfig.project.ejected) {
       throw new SilentError('An ejected project cannot use the build command anymore.');
     }
-    rimraf.sync(path.resolve(this.project.root, outputPath));
+
+    const contentBase = path.resolve(this.project.root, outputPath);
+    if (serveTaskOptions.useDist) {
+      if(!fs.existsSync(contentBase)) {
+        throw new SilentError('Dist directory not found');
```

Comment:
`build` deletes the folder if it exists and then creates it again. It'd make sense for `serve` to act the same. Just having old folder left from `build` or earlier `serve` shouldn't break running `serve` again.